app/models/modal-window.ts

```diff
@@ -10,10 +10,16 @@
 export interface ModalInfo {
     readonly id: number;
     readonly selector: string;
+    readonly type: ModalType
     data: any;
 }
 
+export enum ModalType {
+    MODAL, // Will display as a drop down on macOS
+    CHILD, 
+} 
+
 export interface ModalService {
-    createModal(selector: string): [ModalInfo, Promise<any>];
+    createModal(selector: string, type: ModalType): [ModalInfo, Promise<any>];
```

Comment:
Here when I use the any inside the Promise, because the any comes from an IPC call, there's no way to guarantee it'll be a certain type and I cast it to what I expect it to be.

Going forward I think we use Promises as expected and hope Typescript improves in this area